src/main/java/com/example/springbootrestexample/logic/BookLogic.java

```diff
@@ -0,0 +1,81 @@
+package com.example.springbootrestexample.logic;
+
+import com.example.springbootrestexample.comparators.BookIsbnComparator;
+import com.example.springbootrestexample.comparators.BookNameComparator;
+import com.example.springbootrestexample.exceptions.BookIsbnDoesNotExistException;
+import com.example.springbootrestexample.models.Book;
+import com.example.springbootrestexample.repositories.AuthorRepository;
+import com.example.springbootrestexample.repositories.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class BookLogic extends ExceptionMessageLogic {
+
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private AuthorRepository authorRepository;
+
+    public List<Book> getAllBooks(){
+        return getAllBooks("");
+    }
+
+    public List<Book> getAllBooks(String columnOrder){
+        List<Book> books = bookRepository.listBooks();
+
+        switch (columnOrder){
+            case "isbn":
+                books.sort(new BookIsbnComparator());
+                break;
+
+            case "name":
+                books.sort(new BookNameComparator());
+                break;
+
+            default:
+                break;
+        }
+
+        return books;
+    }
+
+    public Book getBookByIsbn(String isbn){
+        try {
+            return bookRepository.getBookByIsbn(isbn);
+        } catch (NoSuchElementException e) {
+            throw new BookIsbnDoesNotExistException(isbn, e);
+        }
+    }
+
+    public boolean getBookExistByIsbn(String isbn){
+        return (bookRepository.getBookByIsbn(isbn)!= null);
+    }
+
+    public Book createBook(Book book){
```

Comment:
The exception could be more specific already here, there is enough information from the book

Reply:
Do you want me to expose the information of the Exception?

Reply:
Nono, just saying to take into account that here exceptions can also be managed, but in this model, those are managed at controllers layer. Both solution are ok